src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -164,11 +164,21 @@ protected function refreshUser(TokenInterface $token)
         $userDeauthenticated = false;
 
         foreach ($this->userProviders as $provider) {
+            
+            $providerClass = \get_class($provider);
             if (!$provider instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('User provider "%s" must implement "%s".', \get_class($provider), UserProviderInterface::class));
+                throw new \InvalidArgumentException(sprintf('User provider "%s" must implement "%s".', $providerClass, UserProviderInterface::class));
             }
 
             try {
+                $userClass = \get_class($user);
+                if( !$provider->supportsClass($userClass) )
```

Comment:
I would move this outside the `try ... catch` block. IMO there is no need to use an exception for this. We can instead just continue:

```php
foreach ($this->userProviders as $provider) {
    if (!$provider->supportsClass(\get_class($user)) {
        continue;
    }

    // ...
}
```